packages/playwright-file-snapshots/src/matchers/file-matcher.ts

```diff
@@ -25,39 +36,86 @@ interface MatchValidationFileParams {
   matcherState: ExpectMatcherState;
 }
 
-export function matchValidationFile(
+export async function matchValidationFile(
   params: MatchValidationFileParams,
-): MatcherReturnType {
+): Promise<MatcherReturnType> {
   const { matcherName, actual, serializer, config, options, matcherState } =
     params;
-  const { isNot, utils } = matcherState;
+  const { timeout, isNot, utils } = matcherState;
 
   if (isNot) {
     throw new Error("Matcher negation is not supported");
   }
 
   const { validationDir, outputDir, filterSteps } = config;
-  const { titlePath, testPath } = parseTestInfo(test.info(), filterSteps);
-  let pass: boolean;
+  const { titlePath, testPath, updateSnapshots } = parseTestInfo(
+    test.info(),
+    filterSteps,
+  );
 
   const { name, namingStrategy } = options;
-  const matcherResult = new ValidationFileMatcher({
+  const matcher = new ValidationFileMatcher({
     validationDir,
     outputDir,
     testPath,
     titlePath,
     name,
     namingStrategy,
     serializer,
-  }).matchFileSnapshot(actual);
-  matcherResult.writeFileSnapshots();
-
-  try {
-    baseExpect(matcherResult.actual).toBe(matcherResult.expected);
-    pass = true;
-  } catch {
-    pass = false;
+  });
+
+  const testTimeout = options.timeout ?? timeout;
+  const updateDelay = Math.min(testTimeout, UPDATE_DELAY);
+  const snapshot = createSnapshotInstance(actual);
+
+  const isUpdate = matcher.isValidationFileMissing || updateSnapshots;
+  if (isUpdate && snapshot instanceof RepeatableSnapshot) {
+    await waitForTimer(updateDelay);
```

Comment:
This makes the matcher wait for the "first" delay when the validation file is missing, but not when it exists and it should be verified. Is that intended?

Reply:
By now, I saw you mention this explicitly in the README. I guess this is fine.

I still feel like a two-consecutive-snapshots-must-match approach might be better, but it is much more complex...

Reply:
I tried this approach, but figured it does not really add value. You should always take the last snapshot as validation file, because this one is most likely the stable one. So instead of retrying one or more times, which wastes time, you can use an upfront delay before taking the snapshot.

Should the default values not work for all scenarios, we can provide explicit options for overriding them. But for now, I wanted to keep the API simple.